### ISSUE TITLE
Missing resources in Maven tutorial, web app section

### ISSUE THREAD

**Issue by jrao** [osgi/osgi.enroute.site#129]
I worked through the base enRoute tutorial and most of the Maven tutorial up to the end of the web app section: http://enroute.osgi.org/tutorial_eval/450-web. Excellent tutorials! But at the end of the web app section, I ran into an issue:

First of all, there's an inconsistency about where to add the home.htm and about.htm files. The Static Web Content section near the top of the tutorial shows them at this location: osgi.enroute.examples.eval/htm/. However, the actual instructions direct to add these to a different location: osgi.enroute.examples.eval/main/htm/

I started off by following the instructions (using the osgi.enroute.examples.eval/main/htm/ location) but ran into a 404 error. And the evaluator was not present on the home page.

Then I moved home.htm and about.htm to osgi.enroute.examples.eval/htm/ and did not get any 404 but the evaluator was still not present on the home page.

Then I tried copying home.htm and about.htm to both locations and the application worked correctly.

So I'm wondering which is the correct location? And how to get the application to work correctly with these resources only in one of the locations?

**Comment by jrao** [comment]
To reproduce the issue, see my project here: https://github.com/jrao/osgi.enroute.examples.eval

It has the home.htm and about.htm in the osgi.enroute.examples.eval/main/htm/ location and the 404 occurs.

**Comment by pkriens** [comment]
It is htm/main/home.htm|about.htm. I guess the ‘main’ in the Maven like project structure tripped me up. The url to the app is http://localhost:8081/osgi.enroute.examples.eval/index <http://localhost:8081/osgi.enroute.examples.eval/index>.html

I’ve updated the tutorial to reflect the correct paths. 

Let me know if this works now for you. Kind regards,

	Peter Kriens

**Comment by jrao** [comment]
Thanks for looking into this. I updated my app to use the correct paths but I'm still getting the 404. The index.html loads but the home.htm and about.htm fragments aren't found. This is the same behavior as before.

See https://github.com/jrao/osgi.enroute.examples.eval

**Comment by pkriens** [comment]
I looked into it and the problem is caused by the index.html file. This file needs to be preprocessed by bnd but it looks like it is overwritten in a module build by maven. If this file is not preprocessed, then the runtime does not know from which bundle the request came and cannot find the proper resources.

The quick fix is to move the index.html to the root of the project from and then change the bnd.bnd file in the application project from:

-includeresource: \
   {static/osgi.enroute.examples.eval/index.html=\
   	src/main/resources/static/osgi.enroute.examples.eval/index.html}
TO:

-includeresource: \
   {static/osgi.enroute.examples.eval/index.html=\
   	index.html}

This has something to do with ordering in maven and it might have been caused by some of the recent changes. Due to timing the tutorial used a snapshot of bnd, which is still not released.

I will change the tutorial accordingly, sorry for the inconvenience.

Kind regards,

	Peter Kriens
